.scripts/build.sh

```diff
@@ -3,36 +3,174 @@
 set -e
 
 ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")" && cd .. && pwd)
+THEMES="${ROOT}/wp-content/themes/*/"
 LIVE=0
-
 OPT_COMPOSER=""
 
-RED='\e[0;31m'
-GREEN='\e[0;32m'
-YELLOW='\e[0;33m'
-NC='\e[0m' # No Color
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
 
 # SETUP AND SANITY CHECKS
 # =======================
 while getopts l OPTION 2>/dev/null; do
-  case $OPTION
-  in
-    l) LIVE=1;;
-  esac
+    case $OPTION
+        in
+        l) LIVE=1;;
+    esac
 done
-
 if [ $LIVE -eq 1 ]; then
-  OPT_COMPOSER="--no-dev --optimize-autoloader"
+    OPT_COMPOSER="--no-dev --optimize-autoloader"
 fi
 
-echo -e "${GREEN}Installing Composer dependencies…${NC}"
-composer install --no-interaction $OPT_COMPOSER
+# FUNCTIONS DECLARATIONS
+# =======================
+fetch_git(){
+  REPOSITORY=$1
+  NAME=$2
+  TARGETSUB=$3
+  TARGETDIR="${ROOT}${TARGETSUB}${NAME}"
+  if [ ! -z "${REPOSITORY}" ]; then
+    while true; do
+      read -p "Would you like to clone or install as submodule (c/s)? " yn
+      case $yn in
+          [Cc]* )
+            mkdir $TARGETDIR
+            git clone $REPOSITORY $TARGETDIR
+            sleep 3
+            break;;
+          [Ss]* )
+            echo "#Excluding submodule '${NAME}'\n!$TARGETSUB${NAME}" >> .gitignore
+            git submodule add $REPOSITORY ".${TARGETSUB}${NAME}"
+            sleep 3
+            break;;
+          * ) echo "Please answer c or s.";;
+      esac
+    done
+  else
+    echo "${RED}Trying to fetch an invalid git url. exiting...${NC}"
+    exit
+  fi
+}
 
-echo -e "${GREEN}Installing NPM dependencies…${NC}"
-npm install
+fetch_theme(){
+  while true; do
+    read -p "Would you like to install a THEME? " yn
+    case $yn in
+      [Yy]* )
+        shopt -s nullglob
+        WPTHEMES=('git repository URL');
+        WPTHEMES+=($(ls -d $THEMES | xargs -n1 basename))
+        echo "Please select the theme to install:"
+        select opt in "${WPTHEMES[@]}"
+        do
+          if [[ $opt = 'git repository URL' ]]; then
+            read -p "Theme git repository URL: " REPOSITORY
+            THEMENAME=$(basename "$REPOSITORY")
+            THEMENAME="${THEMENAME%.*}" #removing extension
+            fetch_git $REPOSITORY $THEMENAME '/wp-content/themes/'
+            build_theme $THEMENAME
+          else
+            build_theme $opt
+          fi
+        done;;
+      [Nn]* ) break;;
+      * ) echo "Please answer yes or no.";;
+    esac
+  done
+}
+
+build_theme(){
+  THEMENAME=$1
+  while true; do
+    read -p "Would you like to build '${THEMENAME}'? " yn
+    case $yn in
+      [Yy]* )
+        THEMEDIR="$ROOT/wp-content/themes/$THEMENAME"
+        echo  "${GREEN}Building $1 theme…${NC}"
+        cd "${THEMEDIR}"
+        composer install --no-interaction $OPT_COMPOSER
+        npm install
```

Comment:
Isso é apenas uma sugestão: Poderia trocar para o yarn.

Reply:
That is a nice suggestion. Added to roadmap.